src/server/server_family.cc

```diff
@@ -2839,6 +2839,10 @@ void ServerFamily::ReplTakeOver(CmdArgList args, ConnectionContext* cntx) {
 }
 
 void ServerFamily::ReplConf(CmdArgList args, ConnectionContext* cntx) {
+  if (!ServerState::tlocal()->is_master) {
```

Comment:
I think that this should be checked with the replicaof_mu_ locked

Reply:
Why?
It's a thread-local variable, so it can't be read non-atomically.
Also, there are many places that use it, all without that lock. Furthermore, the lock is only available in ServerFamily, while other places use it as well..?

Reply:
the is_master is eventually consistent. It can be that the server is currenly configured as a replica and did not get to set the is_master flag on this thread right?

Reply:
Sure, updated :)